resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -82,6 +82,12 @@ private[spark] class KubernetesClusterSchedulerBackend(
           KUBERNETES_EXECUTOR_ANNOTATION_PREFIX,
           KUBERNETES_EXECUTOR_ANNOTATIONS,
           "executor annotation")
+  private val nodeSelector =
+      ConfigurationUtils.combinePrefixedKeyValuePairsWithDeprecatedConf(
```

Comment:
`ConfigurationUtils.parseKeyValuePairs` only parses the deprecated config format. It's probably sufficient to use `SparkConf.getAllWithPrefix` instead of using anything in `ConfigurationUtils`.